src/main/kotlin/com/github/aleksandrsl/intellijbrowserslist/BrowserslistFoldingBuilder.kt

```diff
@@ -1,48 +1,52 @@
 import com.github.aleksandrsl.intellijbrowserslist.psi.impl.BrowserslistSectionImpl
 import com.intellij.lang.ASTNode
 import com.intellij.lang.folding.FoldingBuilderEx
 import com.intellij.lang.folding.FoldingDescriptor
 import com.intellij.openapi.editor.Document
 import com.intellij.openapi.editor.FoldingGroup
 import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.util.TextRange
 import com.intellij.psi.PsiElement
 import com.intellij.psi.util.PsiTreeUtil
-import com.intellij.psi.util.descendantsOfType
-import com.intellij.util.containers.map2Array
+import com.intellij.psi.util.endOffset
+import com.intellij.psi.util.startOffset
+
 
 class BrowserslistFoldingBuilder : FoldingBuilderEx(), DumbAware {
 
     override fun buildFoldRegions(
-        root: PsiElement,
-        document: Document,
-        quick: Boolean
+        root: PsiElement, document: Document, quick: Boolean
     ): Array<FoldingDescriptor> {
-        val sections = PsiTreeUtil.findChildrenOfType(root, BrowserslistSectionImpl::class.java).filter {
-            it.sectionQueries !== null
+        val sections = PsiTreeUtil.findChildrenOfType(root, BrowserslistFullSection::class.java).filter {
+            it.queryExpressionList.isNotEmpty()
         }
 
-        return sections.map2Array {
-            FoldingDescriptor(
-                it.node,
-                // Last element is usually EOL, if we don't skip it the next line is glued to the folding text.
-                // Still could be improved on the parser side. We don't want to include comments
-                // that are not followed by any query to the section. But this
-                // could require look ahead. Which is not great or maybe possible.
-                it.sectionQueries!!.textRange.grown(-1),
-                FoldingGroup.newGroup(it.sectionHeader.text)
-            )
-        }
+        return sections.mapNotNull { section ->
+            val startOffset = section.queryExpressionList.first().startOffset
+            val endOffset = section.queryExpressionList.last().endOffset
+            if (endOffset > startOffset) {
+                FoldingDescriptor(
+                    section.node,
+                    // Still could be improved on the parser side. We don't want to include comments
+                    // that are not followed by any query to the section. But this
+                    // could require look ahead. Which is not great or maybe possible.
+                    TextRange(startOffset, endOffset), FoldingGroup.newGroup(section.sectionName)
+                )
+            } else {
+                null
+            }
+        }.toTypedArray()
     }
 
     override fun getPlaceholderText(node: ASTNode): String {
         val psi = node.psi
         if (psi is BrowserslistSectionImpl) {
-            val queriesNumber = psi.sectionQueries?.descendantsOfType<BrowserslistQuery>()?.count() ?: return "..."
+            val queriesNumber = psi.queryExpressionList.size
```

Comment:
I think I was doing `[name] ... 2 queries` before, now it's 
```
[name] 
2 queries
```

So I don't think ... is required